tsl/src/data_node.c

```diff
@@ -317,13 +306,20 @@ create_hypertable_data_node_datum(FunctionCallInfo fcinfo, HypertableDataNode *n
 }
 
 static List *
-create_data_node_options(const char *host, int32 port, const char *dbname, const char *user)
+create_data_node_options(const char *host, int32 port, const char *dbname, const char *user,
+						 const char *password)
 {
 	DefElem *host_elm = makeDefElem("host", (Node *) makeString(pstrdup(host)), -1);
 	DefElem *port_elm = makeDefElem("port", (Node *) makeInteger(port), -1);
 	DefElem *dbname_elm = makeDefElem("dbname", (Node *) makeString(pstrdup(dbname)), -1);
 	DefElem *user_elm = makeDefElem("user", (Node *) makeString(pstrdup(user)), -1);
 
+	if (NULL != password)
```

Comment:
nit: could reduce nesting (unless you are using block to define var)

``` C
if (password == NULL)
    return list_make4(host_elm, port_elm, dbname_elm, user_elm);
```

Reply:
Yeah, I didn't want to define the password DefElm unless password was non-NULL.